packages/cardano-services-client/src/HandleProvider/KoraLabsHandleProvider.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Cardano,
+  HandleProvider,
+  HandleResolution,
+  HealthCheckResponse,
+  NetworkInfoProvider,
+  ProviderError,
+  ProviderFailure,
+  ResolveHandlesArgs
+} from '@cardano-sdk/core';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { IHandle } from '@koralabs/handles-public-api-interfaces';
+import axios, { AxiosAdapter, AxiosInstance } from 'axios';
+
+/**
+ * The KoraLabsHandleProvider endpoint paths.
+ */
+const paths = {
+  handles: '/handles',
+  healthCheck: '/health'
+};
+
+export interface KoraLabsHandleProviderDeps {
+  serverUrl: string;
+  networkInfoProvider: NetworkInfoProvider;
+  adapter?: AxiosAdapter;
+  policyId: Cardano.PolicyId;
+}
+
+export const toHandleResolution = ({
+  apiResponse,
+  tip,
+  policyId
+}: {
+  apiResponse: IHandle;
+  tip: Cardano.Tip;
+  policyId: Cardano.PolicyId;
+}): HandleResolution => ({
+  handle: apiResponse.name,
+  hasDatum: apiResponse.hasDatum,
+  policyId,
+  resolvedAddresses: {
+    cardano: Cardano.PaymentAddress(apiResponse.resolved_addresses.ada)
+  },
+  resolvedAt: {
+    hash: tip.hash,
+    slot: tip.slot
+  }
+});
+
+/**
+ * Creates a KoraLabs Provider instance to resolve Standard Handles
+ *
+ * @param KoraLabsHandleProviderDeps The configuration object fot the KoraLabs Handle Provider.
+ */
+export class KoraLabsHandleProvider implements HandleProvider {
+  private axiosClient: AxiosInstance;
+  private networkInfoProvider: NetworkInfoProvider;
+  policyId: Cardano.PolicyId;
+
+  constructor({ serverUrl, networkInfoProvider, adapter, policyId }: KoraLabsHandleProviderDeps) {
+    this.networkInfoProvider = networkInfoProvider;
+    this.axiosClient = axios.create({
+      adapter,
+      baseURL: serverUrl
+    });
+    this.policyId = policyId;
+  }
+
+  async resolveHandles(args: ResolveHandlesArgs): Promise<Array<HandleResolution | null>> {
+    try {
+      const tip = await this.networkInfoProvider.ledgerTip();
+      const results = await Promise.all(
```

Comment:
Good catch @VanessaPC 💪 

> the moment a handle comes as **null**, it won't resolve further handles.

Do you mean _Not found_, i.e a `404` HTTP response, which `axios` translates to rejecting with an error? If so, how does that not:

> ...affect the implementation on the txBuilder or outputBuilder.

... if a single error would result in a rejection of the batch? Maybe you can walk me through this on a call.

Reply:
Oh, I'm guessing it's because we're only resolving a single handle at a time. If that's the case, I agree we can defer this. Please add a ticket to the backlog.

Reply:
Yea, currently we are only using it to resolve a single handle in the OutputBuilder, so for now it's not going to break anything. But since the HandleProvider is in a separate package, it may potentially use elsewhere and it wouldn't work because it's implementation doesn't handle batch requests, where one of the handles may return a 404, but the rest should still resolve, also, because this is what we say we do on the interface, ultimately, we return an array of resolved handles. 

Ticket: https://input-output.atlassian.net/browse/LW-6778

Reply:
Thanks! The follow-up task is not critical, but we should aim to cover it off in the next sprint, since it is a bug. BTW, as this is an open source repo, please only post a reference to Jira tickets i.e `LW-6778`